ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixParticipantTest.java

```diff
@@ -63,19 +65,69 @@
     props.setProperty("clustermap.dcs.zk.connect.strings", zkJson.toString(2));
     helixManagerFactory = new MockHelixManagerFactory();
   }
 
+  private ReplicaId createMockReplicaId(String partitionIdString) {
+    ReplicaId replicaId = Mockito.mock(ReplicaId.class);
+    PartitionId partitionId = Mockito.mock(PartitionId.class);
+    Mockito.when(partitionId.toPathString()).thenReturn(partitionIdString);
+    Mockito.when(replicaId.getPartitionId()).thenReturn(partitionId);
+    return replicaId;
+  }
+
+  /**
+   * Tests setReplicaSealedState method for {@link HelixParticipant}
+   * @throws IOException
+   */
+  @Test
+  public void testSetReplicaSealedState() throws IOException {
+    //setup HelixParticipant and dependencies
+    String partitionId = "somePartitionId";
+    ReplicaId replicaId = createMockReplicaId(partitionId);
+    String hostname = "localhost";
+    int port = 2200;
+    String instanceName = ClusterMapUtils.getInstanceName(hostname, port);
+    HelixParticipant helixParticipant =
+        new HelixParticipant(new ClusterMapConfig(new VerifiableProperties(props)), helixManagerFactory);
+    helixParticipant.initialize(hostname, port, Collections.EMPTY_LIST);
+    HelixManager helixManager = helixManagerFactory.getZKHelixManager(null, null, null, null);
+    HelixAdmin helixAdmin = helixManager.getClusterManagmentTool();
+    InstanceConfig instanceConfig = new InstanceConfig("someInstanceId");
+    helixAdmin.setInstanceConfig(clusterName, instanceName, instanceConfig);
+    instanceConfig = helixAdmin.getInstanceConfig(clusterName, instanceName);
+    InstanceConfig ic = helixAdmin.getInstanceConfig(clusterName, instanceName);
+
+    //Make sure the current sealedReplicas list is null
+    List<String> sealedReplicas = ClusterMapUtils.getSealedReplicas(instanceConfig);
+    assertTrue("sealedReplicas is not null", sealedReplicas == null);
+
+    //Check that invoking setReplicaSealedState adds the partition to the list of sealed replicas
+    helixParticipant.setReplicaSealedState(replicaId, true);
+    sealedReplicas = ClusterMapUtils.getSealedReplicas(helixAdmin.getInstanceConfig(clusterName, instanceName));
+    assertFalse("sealedReplicas is null", sealedReplicas == null);
+    assertTrue("sealedReplicas is not size 1, it's size " + sealedReplicas.size(), sealedReplicas.size() == 1);
+    assertTrue(sealedReplicas.get(0) + " does not equal '" + partitionId + "'",
+        sealedReplicas.get(0).equals(partitionId));
+
+    //Check that invoking setReplicaSealedState with isSealed == false removes partition from list of sealed replicas
+    helixParticipant.setReplicaSealedState(replicaId, false);
+    sealedReplicas = ClusterMapUtils.getSealedReplicas(helixAdmin.getInstanceConfig(clusterName, instanceName));
+    assertFalse("sealedReplicas is null", sealedReplicas == null);
+    assertTrue("sealedReplicas is not size 0, it's size " + sealedReplicas.size(), sealedReplicas.size() == 0);
```

Comment:
Other tests that might be useful: tests to add more than once (can that happen? is that a bug? could happen if we are also using a tool or something I guess). tests to remove more than once.